contracts/src/internal/HyperdriveCheckpoint.sol

```diff
@@ -315,6 +325,23 @@ abstract contract HyperdriveCheckpoint is
             lpSharePrice
         );
 
+        // Claim the checkpoint reward on behalf of the sender.
+        //
+        // NOTE: We do this in a low-level call and ignore the status to ensure
+        // that the checkpoint will be minted regardless of whether or not the
+        // call succeeds.
+        if (_checkpointRewarder != address(0)) {
+            bool isTrader = _isTrader; // avoid stack-too-deep
+            (bool _success, ) = _checkpointRewarder.call(
```

Comment:
weren't you going to wrap this in a try-catch?

Reply:
`try-catch` doesn't catch all errors. This issue goes into some details: https://github.com/ethereum/solidity/issues/12725. The issue mentioned in the original post doesn't pertain to this, but there are other issues that can also result in reverts. Doing a low-level call is (unfortunately) safer.